src/message_element_metadata_owner.cc

```diff
@@ -120,37 +125,58 @@ namespace grpc_labview {
             return;
         }    
         int clusterOffset = 0;
+        int maxAlignmentRequirement = 0;
         for (auto element: metadata->_elements)
         {
-            auto elementType = element->type;
-            auto nestedElement = element;
-            while (elementType == LVMessageMetadataType::MessageValue)
-            {
-                auto nestedMetadata = FindMetadata(element->embeddedMessageName);
-                auto nestedElement = nestedMetadata->_elements.front();
-                elementType = nestedElement->type;
-            }    
-            clusterOffset = AlignClusterOffset(clusterOffset, element->type, element->isRepeated);
-            element->clusterOffset = clusterOffset;
             if (element->type == LVMessageMetadataType::MessageValue)
-            {                
+            {
                 auto nestedMetadata = FindMetadata(element->embeddedMessageName);
                 UpdateMetadataClusterLayout(nestedMetadata);
+                int alignmentRequirement = 0;
+                int elementSize = 0;
                 if (element->isRepeated)
-                {                
-                    clusterOffset += ClusterElementSize(element->type, element->isRepeated);
+                {
+                    elementSize = ClusterElementSize(element->type, element->isRepeated);
+                    // This should be the alignmentRequirement of the LV1DArray struct which should be 4 (contains int32 and int8)
+                    alignmentRequirement = 4;
                 }
                 else
                 {
-                    clusterOffset += nestedMetadata->clusterSize;
+                    alignmentRequirement = nestedMetadata->alignmentRequirement;
+                    elementSize = nestedMetadata->clusterSize;
+                }
+                clusterOffset = AlignClusterOffset(clusterOffset, alignmentRequirement);
+                element->clusterOffset = clusterOffset;
+                clusterOffset += elementSize;
+                if (maxAlignmentRequirement < alignmentRequirement)
+                {
+                    maxAlignmentRequirement = alignmentRequirement;
                 }
             }
             else
             {
-                clusterOffset += ClusterElementSize(element->type, element->isRepeated);
+                clusterOffset = AlignClusterOffset(clusterOffset, element->type, element->isRepeated);
+                element->clusterOffset = clusterOffset;
+                int alignmentRequirement = 0;
+                auto elementSize = ClusterElementSize(element->type, element->isRepeated);
+                clusterOffset += elementSize;
+                if (element->isRepeated)
+                {
+                    // This should be the alignmentRequirement of the LV1DArray struct which should be 4 (contains int32 and int8)
+                    alignmentRequirement = 4;
```

Comment:
We might have to do something similar for Strings and BytesValue as well since they will also be struct of an int32 count and an int8 single-element array. So the size of the struct will be 8 but byte alignment should be 4.

Reply:
I had missed the fact that for strings/bytearrays we would have a pointer to the LStr here and not the actual struct itself. So this would be 8 on 64-bit. I have fixed this assumption for arrays as well.